### PR TITLE
Store log receipts

### PR DESCRIPTION
**Pull request by pablomendezroyo**
Store log receipts in reth by default and allow it to be configurable

### CODE REVIEW

File: reth/entrypoint.sh [hunk 1]
```diff
@@ -31,4 +31,9 @@ exec reth \
   --rpc.max-blocks-per-filter="${RPC_MAX_BLOCKS_PER_FILTER}" \
   --authrpc.addr 0.0.0.0 \
   --authrpc.port 8551 \
-  --authrpc.jwtsecret "${JWT_PATH}" ${EXTRA_OPTS}
+  --authrpc.jwtsecret "${JWT_PATH}" \
+  --block-interval "${BLOCK_INTERVAL}" \
+  $([ "${PRUNE_SENDERRECOVERY_FULL}" = true ] && printf -- "--prune.senderrecovery.full") \
+  --prune.receipts.before "${PRUNE_RECEIPTS_BEFORE}" \
+  --prune.accounthistory.distance "${PRUNE_ACCOUNTHISTORY_DISTANCE}" \
+  --prune.storagehistory.distance "${PRUNE_STORAGEHISTORY_DISTANCE}" ${EXTRA_OPTS}
```
Comment (3alpha) [inline reth/entrypoint.sh:39]: None of these flags should be present when archive mode is set to true.

File: reth/entrypoint.sh [hunk 2]
```diff
@@ -13,7 +13,7 @@ echo "[INFO - entrypoint] Running Reth client for network: ${NETWORK}"
 # shellcheck disable=SC2086
 exec reth \
   node \
-  $( [ "${ARCHIVE_NODE}" = false ] && printf -- "--full" ) \
+  $([ "${ARCHIVE_NODE}" = false ] && printf -- "--full") \
```
Comment (3alpha) [inline reth/entrypoint.sh:16]: We shouldn't have `--full` flag in any case, prune flags are for that.

File: docker-compose.yml [hunk 3]
```diff
@@ -11,6 +11,11 @@ services:
       - reth:/data/reth
     environment:
       RPC_MAX_BLOCKS_PER_FILTER: "0"
+      BLOCK_INTERVAL: "0"
```
Comment (3alpha) [inline docker-compose.yml:14]: Why 0?

Comment (pablomendezroyo) [reply docker-compose.yml:14]: set to 5

**Comment by easeev** [comment]
> allow it to be configurable

This part ended up missing though?

**Pull request merged**